code/game/objects/items/apc_frame.dm

```diff
@@ -6,8 +6,8 @@
 	item_state = "syringe_kit"
 	w_class = WEIGHT_CLASS_SMALL
 	var/result_path
-	var/inverse = 0
-	// For inverse dir frames like light fixtures.
+	var/inverse = 0 // For inverse dir frames like light fixtures.
+	var/onto_wall = FALSE //If the object places onto the wall instead of the floor
```

Comment:
that's why the ndir is passed into New/Initialize, so you know the dir the player was facing at the time, which is the dir you need to shift in (usually it's 24 pixels, but idk about intercoms)